src/server/repository/public/user/getOne.ts

```diff
@@ -28,15 +28,29 @@ export const getOne = async (
     throw new Error('Missing parameter')
   }
 
-  return client.oneOrNone<User | undefined>(
-    `
+  return client
+    .oneOrNone<User | undefined>(
+      `
         select ${selectFields}, jsonb_agg(to_jsonb(ur.*)) as roles
         from public.users u
         left join users_role ur on u.id = ur.user_id
         ${where}
         group by ${selectFields}
     `,
-    [value],
-    Objects.camelize
-  )
+      [value]
+    )
+    .then((data) =>
+      data
+        ? {
+            ...Objects.camelize(data),
+            roles:
+              data.roles && data.roles[0] !== null
```

Comment:
why first item should be not null? is this meant to be length > 0 ?

Reply:
nope, this is a getOne, it doesn't return an array. I added this check because otherwise the tests fail. but I've made some changes to make it more readable.

<img width="834" alt="Schermata 2022-09-30 alle 10 17 20" src="https://user-images.githubusercontent.com/10047945/193226573-3936ebad-7cad-4053-b3b9-8202364a9e8c.png">